File: cpp/3084.cpp
class Solution {
public:
    long long countSubstrings(string& s, char c) {
        long long n = 0;
        for (int l = s.size() - 1; l >= 0; l--)
            if (s[l] == c)
                n++;
        return (n * (n + 1)) / 2;
    }
};

File: java/3084.java
class Solution {
    public long countSubstrings(String s, char c) {
        long n = 0;
        for (int l = s.length() - 1; l >= 0; l--)
            if (s.charAt(l) == c)
                n++;
        return (n * (n + 1)) / 2;
    }
}
